### PR TITLE
New Provider Profile UI + ProviderViewModel refactoring

### LINKED ISSUE #415: Redesign the Provider profile screen

### PR DESCRIPTION
**Pull request by manunam2002**
### Description

This PR adresses the logic & efficiency problem caused by the usage of `ListProviderViewModel` in the Provider profile UI, instead of a dedicated ViewModel for a single profile. It also adresses the UI problems in the `ProviderProfileScreen` by revisiting the style and using a similar theme as in the `SeekerProfileScreen`.

### Changes Made
- `ProviderViewModel` & `ProviderRepository` : Created the necessary functions to use this ViewModel for the Provider profile management.
- `ProviderProfileScreen` & `SeekerProfileScreen`: UI and design changes for the Provider Screen, Made some common elements in the Seeker Screen Reusable to avoid code duplication.
- **Testing**: Adapted existing tests and created new one to maintain high  coverage.

### Related Issues

Fixes #415 #419

### CODE REVIEW

File: app/src/main/java/com/android/solvit/provider/model/profile/ProviderViewModel.kt
Comment (yshdb5) [inline app/src/main/java/com/android/solvit/provider/model/profile/ProviderViewModel.kt:1]: Thank you! Finally, a dedicated VM for the provider. This was much needed.

File: app/src/main/java/com/android/solvit/provider/ui/profile/ModifyProviderInformation.kt
Comment (yshdb5) [inline app/src/main/java/com/android/solvit/provider/ui/profile/ModifyProviderInformation.kt:1]: LGTM. The UI still needs some consistency but I know it's not your job.

File: app/src/main/java/com/android/solvit/provider/ui/profile/ProviderProfile.kt
Comment (yshdb5) [inline app/src/main/java/com/android/solvit/provider/ui/profile/ProviderProfile.kt:1]: Wow. Fantastic screen.

File: app/src/main/java/com/android/solvit/provider/ui/profile/ProviderProfile.kt [hunk 1]
```diff
@@ -71,508 +69,258 @@ import com.android.solvit.shared.ui.navigation.Screen
...
+              modifier = Modifier.fillMaxWidth().testTag("AdditionalInfosCard"),
+              colors = CardDefaults.cardColors(containerColor = colorScheme.background),
+              elevation = CardDefaults.cardElevation(defaultElevation = 4.dp)) {
+                Column(
+                    verticalArrangement = Arrangement.spacedBy(8.dp),
+                    modifier = Modifier.padding(16.dp)) {
+                      ProviderItem(
+                          icon = Icons.Default.Build,
+                          text = provider?.service?.name ?: "",
+                          tag = "ServiceItem")
```
Comment (yshdb5) [inline app/src/main/java/com/android/solvit/provider/ui/profile/ProviderProfile.kt:137]: Coud you use the for `Services.format(...)` to better display the provider's service type?

File: app/src/main/java/com/android/solvit/provider/ui/profile/ProviderProfile.kt [hunk 2]
```diff
@@ -71,508 +69,258 @@ import com.android.solvit.shared.ui.navigation.Screen
...
+                      ProviderItem(
+                          icon = Icons.Default.Info,
+                          text = provider?.description ?: "",
+                          tag = "DescriptionItem",
+                          maxLines = 3)
+                      ProviderItem(
+                          icon = Icons.Default.ThumbUp,
+                          drawable = R.drawable.language_icon,
+                          text = provider?.languages?.joinToString { it.name.lowercase() } ?: "",
+                          tag = "LanguagesItem")
```
Comment (yshdb5) [inline app/src/main/java/com/android/solvit/provider/ui/profile/ProviderProfile.kt:151]: Same remark about the formatting of the languages, capitalizing the first letter would improve the display, in my opinion.

File: app/src/main/java/com/android/solvit/provider/ui/profile/ProviderProfile.kt [hunk 3]
```diff
@@ -71,508 +69,258 @@ import com.android.solvit.shared.ui.navigation.Screen
...
+                            for (status in ServiceRequestStatus.entries) {
+                              val size =
+                                  when (status) {
+                                    ServiceRequestStatus.PENDING -> filteredPendingTasks.size
+                                    ServiceRequestStatus.ACCEPTED -> filteredAcceptedTasks.size
+                                    ServiceRequestStatus.SCHEDULED -> filteredScheduledTasks.size
+                                    ServiceRequestStatus.COMPLETED -> filteredCompletedTasks.size
+                                    ServiceRequestStatus.CANCELED -> filteredCanceledTasks.size
+                                    ServiceRequestStatus.ARCHIVED -> filteredArchivedTasks.size
+                                  }
```
Comment (yshdb5) [inline app/src/main/java/com/android/solvit/provider/ui/profile/ProviderProfile.kt:315]: It seems like the number of requests for each status is not updated dynamically. When managing requests in the request dashboard of the provider, the modifications are not reflected in the profile display.

**Review by yshdb5 — APPROVED** [review]
### Overall Comment  
This pull request brings key improvements to the `ProviderViewModel` and related UI components, enhancing the code structure and maintainability. Thank you for the transition to `ProviderViewModel` from `ListProviderViewModel`, it is well executed and it should have been done even sooner!

### Highlights  
1. **UI**: The `ProviderProfile` screen is stunning.
2. **Refactoring**: Shifting from `ListProviderViewModel` to `ProviderViewModel` simplifies the code and ensures better separation of concerns.  
3. **Error Handling**: Logging improves debugging and issue monitoring.  
4. **Readability**: Organizing imports and separating logic improves code clarity and maintainability.  

### Suggestions for Improvement  
1. **Comments**: I have left comments with suggestions for improvements, please take them into consideration before merging.

### Conclusion  
This PR significantly improves the project’s architecture, maintainability and UI. Addressing the little comments I have left will further strengthen this great contribution. Excellent work!

File: app/src/main/java/com/android/solvit/provider/ui/profile/ProviderProfile.kt (hunk 2 above)
Comment (manunam2002) [reply app/src/main/java/com/android/solvit/provider/ui/profile/ProviderProfile.kt:151]: I'll do just that

File: app/src/main/java/com/android/solvit/provider/ui/profile/ProviderProfile.kt (hunk 3 above)
Comment (manunam2002) [reply app/src/main/java/com/android/solvit/provider/ui/profile/ProviderProfile.kt:315]: This issue is caused by the fact that we use different instances of the `ServiceRequestViewModel` in the job dashboard and in the profile screen so the changes are not instantly synced  (this is one of the reason we should use the same instance of the ViewModels). However I fix this in my other PR.

File: app/src/main/java/com/android/solvit/provider/ui/profile/ProviderProfile.kt
Comment (hjamaly) [inline app/src/main/java/com/android/solvit/provider/ui/profile/ProviderProfile.kt:1]: Agree with new design ! Old design was a little bit too complex and not very refined !

File: app/src/main/java/com/android/solvit/provider/ui/profile/ProviderProfile.kt (hunk 3 above)
Comment (hjamaly) [reply app/src/main/java/com/android/solvit/provider/ui/profile/ProviderProfile.kt:319]: 👍 it's fixed !

File: app/src/main/java/com/android/solvit/seeker/ui/profile/SeekerProfile.kt [hunk 4]
```diff
@@ -144,45 +115,8 @@ fun SeekerProfileScreen(
...
-                          }
-                        }
-                        IconButton(
-                            onClick = { navigationActions.navigateTo(Screen.EDIT_PROFILE) },
-                            modifier = Modifier.testTag("EditProfileButton")) {
-                              Icon(Icons.Default.Edit, contentDescription = "Edit Profile")
-                            }
-                      }
-                }
+            ProfileInfoCard(
```
Comment (hjamaly) [inline app/src/main/java/com/android/solvit/seeker/ui/profile/SeekerProfile.kt:118]: Nice avoid of code's duplication

**Review by hjamaly — APPROVED** [review]
### Overview
This PR transitions the ProviderProfileScreen to a dedicated ProviderViewModel, resolving inefficiencies tied to ListProviderViewModel, and introduces a redesigned UI aligned with SeekerProfileScreen. New tests were added, and existing ones updated for robust coverage.

### Strengths

- Logical Refactoring: The shift to ProviderViewModel simplifies the architecture and enhances performance and maintainability.
- UI Enhancements: The new ProviderProfileScreen design is consistent, reusable, and visually appealing.
- Error Handling: Improved logging ensures efficient debugging.
- Testing: Updated and new tests maintain strong coverage.

### Suggestions 
- Try to take a look to fetchProviderById coded in listproviderViewModel to avoid duplication with getProvider

### Conclusion
This PR improves performance, design, and maintainability.  Fantastic work!

**Pull request merged**